apps/web/src/app/(pages)/(dashboard)/(roles)/page.tsx

```diff
@@ -27,8 +40,37 @@ export default function Roles() {
   return (
     <>
       {roles.isSuccess && roles.data.length > 0 && (
-        <div className="flex h-[86dvh] w-full gap-4 divide-x divide-[#474747] pl-4 lg:h-[92dvh]">
+        <div className="flex h-[86dvh] w-full gap-4 divide-x divide-[#474747] pl-4 lg:h-[92dvh] ">
           <div className="w-[28%] gap-3 overflow-auto p-1">
+            <div className="text-right">
+              <DropdownMenu>
+                <DropdownMenuTrigger className="pb-2">
+                  Filter By <ChevronDown className="inline" />
+                </DropdownMenuTrigger>
+                <DropdownMenuContent>
+                  <DropdownMenuLabel className="text-center flex flex-col">
+                    <Button
+                      className={buttonStyle}
+                      onClick={() => setSelectedFilter("newest")}
+                    >
+                      Newest
+                    </Button>
+                    <Button
+                      className={buttonStyle}
+                      onClick={() => setSelectedFilter("oldest")}
+                    >
+                      Oldest
+                    </Button>
+                    <Button
+                      className={buttonStyle}
+                      onClick={() => setSelectedFilter("rating")}
+                    >
+                      Rating
+                    </Button>
```

Comment:
If we add Sort by default to the dropdown we should probably make it an option here so that if somebody selects a sorting option they can go back and select default again